### PR TITLE
better success criteria charts

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* Improving the success criteria chart based on @liunelson's feedback (see attached ticket)
* binning is now done by vegalite rather than our own function

Below:
Screenshot 2024-07-24 at 11 25 22 AM

Above:

Screenshot 2024-07-24 at 11 41 44 AM

### CODE REVIEW

File: packages/client/hmi-client/tests/unit/utils/chart.spec.ts
Comment (blanchco) [inline packages/client/hmi-client/tests/unit/utils/chart.spec.ts:1]: We no longer do the binning here it is done by vegalite so the function has been removed

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-drilldown.vue [hunk 1]
```diff
@@ -253,19 +253,8 @@
...
-											:visualization-spec="
-												createOptimizeChart(
-													riskResults[knobs.postForecastRunId],
-													constraint.targetVariable,
-													constraint.threshold,
-													constraint.isMinimized
-												)
-											"
-										/>
+									<li v-for="(_constraint, key) in node.state.constraintGroups" :key="`success_criteria_${key}`">
```
Comment (Tom-Szendrey) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-drilldown.vue:256]: Whats the point of the :key being a static string concat with the index?

**Review by shawnyama — APPROVED** [review]
lgtm

**Review by Tom-Szendrey — APPROVED** [review]
Thanks

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-drilldown.vue (hunk 1 above)
Comment (blanchco) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-drilldown.vue:256]: I thought it was screwing with other list objects with the same key but thats not even an issue 😆. Its not needed, Ill make it just `key`

**Pull request merged**